### PR TITLE
#315: Implement par template with recursion

### PR DESCRIPTION
**Pull request by paulodamaso**
For #315: Implementation of par templatewith recursion:
- Added documentation to `par.xml`
- Corrected message if test in `par.xml` fails
- Corrected urls to `https` in `par.xml`
- Removed `@todo` from `par.xml`
- Changed `par` template in `templates.xml` to use recursion

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#370](https://github.com/zerocracy/datum/issues/370) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#370](https://github.com/zerocracy/datum/issues/370) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1910369548); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: xsl/templates.xsl [hunk 1]
```diff
@@ -186,7 +186,17 @@ SOFTWARE.
   </xsl:template>
   <xsl:template name="par">
     <xsl:param name="text"/>
-    <xsl:value-of select="$text"/>
+    <xsl:analyze-string select="$text" regex="(§[0-9]*)">
```
Comment (krzyk) [inline xsl/templates.xsl]: @paulodamaso `(§[0-9]*)` -> `(§[0-9]+)`, we expect a paragraph to contain at least one number, not zero or more

Comment (krzyk) [reply xsl/templates.xsl]: @paulodamaso also as you need only the number part, you should move the parentheses after the paragraph character: `§([0-9]+)`

File: xsl/templates.xsl [hunk 2]
```diff
@@ -186,7 +186,17 @@ SOFTWARE.
...
+      <xsl:matching-substring>
+        <xsl:variable select="regex-group(1)" name="paragraph"/>
+        <a href="{concat('https://www.zerocracy.com/policy.html#', substring-after($paragraph,'§'))}">
```
Comment (krzyk) [inline xsl/templates.xsl]: @paulodamaso here you don't need `substring-after`, just `$paragraph`, when you fix the regex above

**Comment by paulodamaso** [comment]
@krzyk Thanks for the comments, corrected the code according to them, please take a look.

**Comment by krzyk** [comment]
@paulodamaso looks good

**Comment by krzyk** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#370](https://github.com/zerocracy/datum/issues/370) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +23 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**